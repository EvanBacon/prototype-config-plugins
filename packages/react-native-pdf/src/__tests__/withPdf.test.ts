import { addAndroidPackagingOptions } from "../withPdf";

describe(addAndroidPackagingOptions, () => {
  it(`adds packaging options to the app build.gradle`, () => {
    const result = addAndroidPackagingOptions(`
      android {
          defaultConfig {
              applicationId 'com.example.app'
          }
      }
  `);
    expect(result.contents).toMatchSnapshot();
    expect(result.didMerge).toBe(true);
    expect(result.didClear).toBe(false);

    const secondResults = addAndroidPackagingOptions(result.contents);

    expect(secondResults.contents).toBe(result.contents);
    expect(secondResults.didMerge).toBe(false);
    expect(secondResults.didClear).toBe(false);
  });
});
