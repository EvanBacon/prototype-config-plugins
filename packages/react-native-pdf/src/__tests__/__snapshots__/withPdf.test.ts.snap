// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAndroidPackagingOptions adds packaging options to the app build.gradle 1`] = `
"
      android {
// @generated begin react-native-pdf-packaging-options - expo prebuild (DO NOT MODIFY) sync-e288c3c5efbb1ef5e1a1959a848260c36558a381

    packagingOptions {
        pickFirst 'lib/x86/libc++_shared.so'
        pickFirst 'lib/x86_64/libjsc.so'
        pickFirst 'lib/arm64-v8a/libjsc.so'
        pickFirst 'lib/arm64-v8a/libc++_shared.so'
        pickFirst 'lib/x86_64/libc++_shared.so'
        pickFirst 'lib/armeabi-v7a/libc++_shared.so'
    }

// @generated end react-native-pdf-packaging-options
          defaultConfig {
              applicationId 'com.example.app'
          }
      }
  "
`;
